src/macaron/database/table_definitions.py

```diff
@@ -1,128 +1,538 @@
+    __table_args__ = (UniqueConstraint("component_id", "requirement_name", name="uq__requirement_name_component_id"),)
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)  # noqa: A003
+
+    #: The software component ID.
+    component_id: Mapped[int] = mapped_column(Integer, ForeignKey("_component.id"), nullable=False)
+
+    #: The unique SLSA requirement name.
+    requirement_name: Mapped[Enum] = mapped_column(
+        Enum(*ReqName._member_names_), nullable=False  # pylint: disable=protected-access,no-member
+    )
+
+    #: The short description of the SLSA requirement.
+    requirement_short_description: Mapped[str] = mapped_column(String, nullable=True)
+
+    #: The justification in the check result for a particular requirement.
+    feedback: Mapped[str] = mapped_column(String, nullable=True)
+
+    #: The many-to-one relationship between SLSA requirements and software components.
+    component: Mapped["Component"] = relationship(back_populates="slsarequirement", lazy="immediate")
+
+
+class CheckResult(ORMBase):
+    """ORM class for the result of a check, is automatically added for each check."""
 
     __tablename__ = "_check_result"
+    # A unique constraint as defined below makes sure a component will have only one result row
+    # for a particular check. Note that this different from a column constraint (unique=True)
+    # that ensures a check ID is unique.
+    # See the naming convention for the declarative metadata (which we don't have currently!):
+    # https://alembic.sqlalchemy.org/en/latest/naming.html#autogen-naming-conventions
+    __table_args__ = (UniqueConstraint("component_id", "check_id", name="uq__check_result_component_id"),)
+
+    #: The primary key.
     id: Mapped[int] = mapped_column(  # noqa: A003 # pylint: disable=invalid-name
         Integer, primary_key=True, autoincrement=True
     )
+
+    #: The check identifier, e.g., mcn_build_as_code_1.
     check_id: Mapped[str] = mapped_column(String, nullable=False)
-    repository: Mapped[int] = mapped_column(Integer, ForeignKey("_repository.id"), nullable=False)
+
+    #: Shows whether a check has passed or failed.
     passed: Mapped[bool] = mapped_column(Boolean, nullable=False)
-    skipped: Mapped[bool] = mapped_column(Boolean, nullable=False)
 
+    #: The foreign key to the software component.
+    component_id: Mapped[int] = mapped_column(Integer, ForeignKey("_component.id"), nullable=False)
 
-class CheckFactsTable:
+    #: A many-to-one relationship with software components.
+    component: Mapped["Component"] = relationship(back_populates="checkresult")
+
+    #: A one-to-many relationship with checks.
+    checkfacts: Mapped[list["CheckFacts"]] = relationship(back_populates="checkresult", lazy="immediate")
+
+
+class CheckFacts(ORMBase):
     """
-    Declarative mixin for check results.
+    This class allows SQLAlchemy to load elements polymorphically, using single table inheritance.
```

Comment:
I think this comment saying `single table inheritance` is not consistent with the implementation (each subclass of `CheckFacts` will have its own table in the database).

Reply:
`single table inheritance` should have been `joined table inheritance`. See commit [374b680](https://github.com/oracle/macaron/pull/305/commits/374b680fa837cb558bbe6e3c16ec2f8a8de75bd9)